external/vcm/vcm/data_transform.py

```diff
@@ -0,0 +1,103 @@
+import dataclasses
+from toolz.functoolz import curry
+from typing import Any, Callable, Literal, Mapping, MutableMapping, Sequence, Set
+import xarray as xr
+import vcm
+
+
+DATA_TRANSFORM_REGISTRY: MutableMapping[str, Mapping[str, Any]] = {}
+
+TransformName = Literal[
+    "q1_from_qm_q2", "qm_from_q1_q2", "q1_from_dQ1_pQ1", "q2_from_dQ2_pQ2",
+]
+
+
+@curry
+def register(
+    name: str,
+    inputs: Sequence[str],
+    outputs: Sequence[str],
+    func: Callable[..., xr.Dataset],
+):
+    if name in DATA_TRANSFORM_REGISTRY:
+        raise ValueError(f"Function {name} has already been added to registry.")
+    DATA_TRANSFORM_REGISTRY[name] = {
+        "func": func,
+        "inputs": inputs,
+        "outputs": outputs,
+    }
+    return func
+
+
+@register("qm_from_q1_q2", ["Q1", "Q2"], ["Qm"])
+def qm_from_q1_q2(ds):
+    ds["Qm"] = vcm.moist_static_energy_tendency(ds["Q1"], ds["Q2"])
+    return ds
+
+
+@register("q1_from_qm_q2", ["Qm", "Q2"], ["Q1"])
+def q1_from_qm_q2(ds):
+    ds["Q1"] = vcm.temperature_tendency(ds["Qm"], ds["Q2"])
```

Comment:
Should this use the air temperature as the input to the latent heat of vaporization? Similarly for the moist static energy tendency, is it calculated using Lv(T at z) or is it the convention to always Lv(T freezing)?

Reply:
originally i had this option as a kwarg, but that doesn't really work because then the required inputs are not well-defined. So I could add a new transform that uses the temperature-dependent version. I previously found it does help a bit in closing MSE budget, but overall is a pretty small factor.